tests/test_node.py

```diff
@@ -213,28 +217,104 @@ def test_owlbot_main(hermetic_mock):
     try:
         os.chdir(temp_dir / "nodejs-dlp")
         # just confirm it doesn't throw an exception.
-        node.owlbot_main(
-            Path(__file__).parent.parent / "synthtool" / "gcp" / "templates"
-        )
+        node.owlbot_main(TEMPLATES)
     finally:
         os.chdir(cwd)
 
 
-@patch("synthtool.languages.node.postprocess_gapic_library_hermetic")
-def test_owlbot_main_with_staging(hermetic_mock):
+@pytest.fixture
+def nodejs_dlp():
+    """chdir to a copy of nodejs-dlp-with-staging."""
     temp_dir = Path(tempfile.mkdtemp())
     shutil.copytree(FIXTURES / "nodejs-dlp-with-staging", temp_dir / "nodejs-dlp")
     cwd = os.getcwd()
     try:
         os.chdir(temp_dir / "nodejs-dlp")
-        # just confirm it doesn't throw an exception.
-        node.owlbot_main(
-            Path(__file__).parent.parent / "synthtool" / "gcp" / "templates"
-        )
+        yield temp_dir / "nodejs-dlp"
     finally:
         os.chdir(cwd)
 
 
+@patch("synthtool.languages.node.postprocess_gapic_library_hermetic")
+def test_owlbot_main_with_staging(hermetic_mock, nodejs_dlp):
+    original_text = open(
+        FIXTURES / "nodejs-dlp-with-staging" / "src" / "index.ts", "rt"
+    ).read()
+    node.owlbot_main(TEMPLATES)
+    # confirm index.ts was overwritten by template-generated index.ts.
+    staging_text = open(
+        FIXTURES
+        / "nodejs-dlp-with-staging"
+        / "owl-bot-staging"
+        / "v2"
+        / "src"
+        / "index.ts",
+        "rt",
+    ).read()
+    text = open("./src/index.ts", "rt").read()
+    assert staging_text != text
+    assert original_text != text
+
+
+@patch("synthtool.languages.node.postprocess_gapic_library_hermetic")
+def test_owlbot_main_with_staging_index_from_staging(hermetic_mock, nodejs_dlp):
+    node.owlbot_main(
+        TEMPLATES,
+        staging_excludes=["README.md", "package.json"],
+        templates_excludes=["src/index.ts"],
+    )
+    # confirm index.ts was overwritten by staging index.ts.
+    staging_text = open(
+        FIXTURES
+        / "nodejs-dlp-with-staging"
+        / "owl-bot-staging"
+        / "v2"
+        / "src"
+        / "index.ts",
+        "rt",
+    ).read()
+    text = open("./src/index.ts", "rt").read()
+    assert staging_text == text
+
+
+@patch("synthtool.languages.node.postprocess_gapic_library_hermetic")
+def test_owlbot_main_with_staging_ignore_index(hermetic_mock, nodejs_dlp):
+    original_text = open(
+        FIXTURES / "nodejs-dlp-with-staging" / "src" / "index.ts", "rt"
+    ).read()
+    node.owlbot_main(TEMPLATES, templates_excludes=["src/index.ts"])
+    # confirm index.ts was overwritten by staging index.ts.
+    text = open("./src/index.ts", "rt").read()
+    assert original_text == text
+
+
+@patch("synthtool.languages.node.postprocess_gapic_library_hermetic")
+def test_owlbot_main_with_staging_patch_staging(hermetic_mock, nodejs_dlp):
+    def patch(library: Path):
```

Comment:
this is clever, I like it. Are you thinking it will work for hermetic replacements?

Reply:
Yep.